addon/instance-initializers/prefetch.js

```diff
@@ -1,103 +1,109 @@
 import { assign } from '@ember/polyfills';
 import { resolve } from 'rsvp';
+import { gte } from 'ember-compatibility-helpers';
 
 export function initialize(instance) {
-  const ROUTER_NAME = 'router:main';
-  const router = (typeof instance.lookup === 'function' ? instance.lookup(ROUTER_NAME) : instance.container.lookup(ROUTER_NAME));
-
-  router.on('willTransition', function(transition) {
-    if (!transition.handlerInfos) {
-      return;
-    }
-
-    const pivotHandler = transition.pivotHandler;
-
-    // If there is no pivot, we should try to prefetch all handlers.
-    let hasSeenPivot = pivotHandler == null ? true : false;
+  if (gte('3.6.0')) {
+    instance.inject('route:application', '__prefetch', 'service:prefetch');
+  } else {
+    // Delete all of this in the Ember 3.8 LTS and rev major
+    const ROUTER_NAME = 'router:main';
+    const router = (typeof instance.lookup === 'function' ? instance.lookup(ROUTER_NAME) : instance.container.lookup(ROUTER_NAME));
```

Comment:
I don't think this ternary is needed anymore. We dropped support for Ember < 2.4 already.